### PR TITLE
Updates to support PPE configuration setting

### PR DESCRIPTION
**Pull request by shoatman**

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/authorities/Environment.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.authorities;
+
+public enum Environment {
+    PreProduction,
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/authorities/Environment.java]: In the event that sovereign ppe environments come along in the future, is it worth naming this something like `PreProductionWorldwide`, `ProductionWorldwide`, etc?

Comment (shoatman) [reply common/src/main/java/com/microsoft/identity/common/internal/authorities/Environment.java]: It looks like clouds belong to an environment (PPE, Production).  At least based on the instance disocovery document which is environment specific.

File: common/src/test/java/com/microsoft/identity/common/unit/AudienceDefaultPreProductionCloudUrlTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    @Test
+    public void test_AnyOrganizationAccountDefaultCloudUrl() {
+
+        AzureActiveDirectoryAudience audience = new AllAccounts();
+        Assert.assertEquals(AzureActiveDirectoryEnvironment.PREPRODUCTION_CLOUD_URL, audience.getCloudUrl());
+
+    }
+
+    @Test
+    public void test_AnyPerosnalAccountDefaultCloudUrl()  {
```
Comment (kreedula) [inline common/src/test/java/com/microsoft/identity/common/unit/AudienceDefaultPreProductionCloudUrlTest.java]: Nit : typo "Personal"

File: common/src/main/java/com/microsoft/identity/common/internal/authorities/AzureActiveDirectoryAudience.java [hunk 3]
```diff
@@ -38,7 +39,11 @@
     public static final String ALL = "common";
 
     public String getCloudUrl() {
-        return mCloudUrl;
+        if(mCloudUrl == null){
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/authorities/AzureActiveDirectoryAudience.java]: Should we check for empty here too?

**Pull request merged**